main/src/cgeo/geocaching/CachePopupFragment.java

```diff
@@ -261,6 +261,22 @@ public void onClick(final View arg0) {
         }
     }
 
+    private class ShowHintClickListener implements View.OnClickListener {
+        @Override
+        public void onClick(final View arg0) {
+            final View popupView = (View) arg0.getParent().getParent();
```

Comment:
Such parent/child traversal code is dangerous, since every change of the layout XML files can break it, and we will not notice it since there is no compile error or similar. Therefore I'm in favor of always using explicit ID based lookups instead of getParent() or getChild() calls.
(I'm aware we have more code with this pattern, nevertheless we should probably at least not write new code using the pattern).

Reply:
I would like to use ButterKnife etc. for doing an ID based search, but what can I use as a root anchor for it to start the search from?

Reply:
fixed with 4e7e23b